### PR TITLE
Vehicle Motion Update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Remove some common code and small cleanup
Continuing on from @hdpoliveira

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -9296,35 +9309,32 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
...
+                    {
+                        spin_speed = 0;
+                    }
                 }
             }
         }
         else
         {
-        loc_6DD054:
             if (abs(velocity) > 0x10000)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp:9338]: Previously, this code also ran when VEHICLE_ENTRY_FLAG_WATER_RIDE was set and vehicle_sprite_type was 00. So the `else` would have to be changed to `if (!(vehicleEntry->flags & VEHICLE_ENTRY_FLAG_WATER_RIDE) || vehicle_sprite_type == 0)`.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp:9338]: I think I'll just copy the code into the above else block instead.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -9296,35 +9309,39 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
...
-                    spin_speed = 0;
+                    if (vehicle_sprite_type == 2)
+                    {
+                        spin_speed = 0;
+                    }
+                }
+            }
+            else
+            {
+                if (abs(velocity) > 0x10000)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp:9337]: Can be condensed into `else if...`

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp]: I've modified it again to make it almost identical to the non golf version getting it ready to combine the two.

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -9296,39 +9309,36 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
...
+                if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
                 {
-                    spin_speed = 0;
+                    if (vehicle_sprite_type == 2)
+                    {
+                        spin_speed = 0;
+                    }
                 }
+                newAcceleration += poweredAcceleration;
+                goto loc_6DD069;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp:9335]: I get why you did this, but I'd prefer if you didn't introduce a goto.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp:9335]: Neither would I but it will be easier to refactor the two together now.

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp:9335]: Do you plan to do that in this PR or in another?

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp:9335]: In another refactor

**Pull request merged**